core/src/main/java/com/sequenceiq/cloudbreak/job/stackpatcher/ExistingStackPatcherJob.java

```diff
@@ -57,36 +61,39 @@ protected void executeTracedJob(JobExecutionContext context) throws JobExecution
         Stack stack = stackService.getByIdWithListsInTransaction(getStackId());
         Status stackStatus = stack.getStatus();
         if (!Status.getUnschedulableStatuses().contains(stackStatus)) {
-            applyStackPatches(stack);
+            String stackPatchTypeName = context.getJobDetail().getJobDataMap().getString(STACK_PATCH_TYPE_NAME);
+            StackPatchType stackPatchType = stackPatchTypeConverter.convertToEntityAttribute(stackPatchTypeName);
+            if (StackPatchType.UNKNOWN.equals(stackPatchType)) {
+                throw new JobExecutionException(String.format("Stack patch type %s converted to UNKNOWN", stackPatchTypeName));
+            }
+            applyStackPatch(stack, stackPatchType);
         } else {
-            LOGGER.debug("Existing stack fixing will be unscheduled, because stack {} state is {}", stack.getResourceCrn(), stackStatus);
+            LOGGER.debug("Existing stack patching will be unscheduled, because stack {} state is {}", stack.getResourceCrn(), stackStatus);
         }
         jobService.unschedule(context.getJobDetail().getKey());
     }
 
-    private void applyStackPatches(Stack stack) throws JobExecutionException {
-        Map<StackPatchType, Exception> errors = new HashMap<>();
-        for (ExistingStackPatchService existingStackPatchService : existingStackPatchServices) {
-            if (!existingStackPatchService.isStackAlreadyFixed(stack)) {
-                try {
-                    applyStackPatch(stack, existingStackPatchService);
-                } catch (ExistingStackPatchApplyException e) {
-                    LOGGER.error("Failed to patch stack: {} for {}", stack.getResourceCrn(), existingStackPatchService.getStackFixType(), e);
-                    errors.put(existingStackPatchService.getStackFixType(), e);
-                }
-            } else {
-                LOGGER.debug("Stack {} was already patched for {}", stack.getResourceCrn(), existingStackPatchService.getStackFixType());
+    private void applyStackPatch(Stack stack, StackPatchType stackPatchType) throws JobExecutionException {
+        ExistingStackPatchService existingStackPatchService = existingStackPatchServices.stream()
+                .filter(s -> stackPatchType.equals(s.getStackPatchType()))
+                .findFirst()
+                .orElseThrow(() -> new JobExecutionException("No stack patcher implementation found for type " + stackPatchType));
```

Comment:
would this result in unscheduling the `StackPatchType` from running into this again and again until a new version is released?

Reply:
It would not unschedule, but this should not happen in prod. Same applies to the other similar question. But I will add an unschedule to them just to be sure.